include/storage/io.hpp

```diff
@@ -144,6 +144,80 @@ class FileReader
     }
 };
 
+class FileWriter
+{
+private:
+	const std::string filename;
+	boost::filesystem::ofstream output_stream;
+
+public:
+	enum FingerprintFlag
+	{
+		GenerateFingerprint,
+		HasNoFingerprint
+	};
+
+
+	FileWriter(const std::string &filename, const FingerprintFlag flag)
+		: FileWriter(boost::filesystem::path(filename), flag)
+	{
+	}
+
+	FileWriter(const boost::filesystem::path &filename_, const FingerprintFlag flag)
+		: filename(filename_.string())
+	{
+		output_stream.open(filename_, std::ios::binary);
+		if (!output_stream)
+			throw util::exception("Error opening " + filename + ":" + std::strerror(errno));
```

Comment:
Yea I generally agree, however, I'd suggest opening another issue for that - at the moment `exception` (https://github.com/Project-OSRM/osrm-backend/blob/master/include/util/exception.hpp) is marked `final`, we'd be better of differentiating exceptions across the board in a refactor.